src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+    public static StrimziKafkaContainer create(final int brokerId) {
+        return new StrimziKafkaContainer(brokerId, Collections.emptyMap());
+    }
+
+    @Override
+    protected void doStart() {
+        // we need it for the startZookeeper(); and startKafka(); to run container before...
+        withCommand("sh", "-c", "while [ ! -f " + STARTER_SCRIPT + " ]; do sleep 0.1; done; " + STARTER_SCRIPT);
+        super.doStart();
+    }
+
+    public StrimziKafkaContainer withExternalZookeeper(final String connectString) {
+        this.externalZookeeperConnect = connectString;
+        return self();
+    }
+
+    @Override
+    protected void containerIsStarting(final InspectContainerResponse containerInfo, final boolean reused) {
+        super.containerIsStarting(containerInfo, reused);
+
+        this.kafkaExposedPort = getMappedPort(KAFKA_PORT);
+
+        LOGGER.info("This is mapped port {}", this.kafkaExposedPort);
+
+        StringBuilder advertisedListeners = new StringBuilder(getBootstrapServers());
+
+        Collection<ContainerNetwork> cns = containerInfo.getNetworkSettings().getNetworks().values();
+
+        int advertisedListenerNumber = 1;
+        List<String> advertisedListenersNames = new ArrayList<>();
+
+        for (ContainerNetwork cn : cns) {
+            // must be always unique
+            final String advertisedName = "BROKER" + advertisedListenerNumber;
+            advertisedListeners.append(",").append(advertisedName).append("://").append(cn.getIpAddress()).append(":9093");
+            advertisedListenersNames.add(advertisedName);
+            advertisedListenerNumber++;
+        }
+
+        LOGGER.info("This is all advertised listeners for Kafka {}", advertisedListeners.toString());
+
+        StringBuilder kafkaListeners = new StringBuilder();
+        StringBuilder kafkaListenerSecurityProtocol = new StringBuilder();
+
+        advertisedListenersNames.forEach(name -> {
+            // listeners
+            kafkaListeners.append(name);
+            kafkaListeners.append("://0.0.0.0:9093");
+            kafkaListeners.append(",");
+            // listener.security.protocol.map
+            kafkaListenerSecurityProtocol.append(name);
+            kafkaListenerSecurityProtocol.append(":PLAINTEXT");
+            kafkaListenerSecurityProtocol.append(",");
+        });
+
+        StringBuilder kafkaConfiguration = new StringBuilder();
+
+        // default listener config
+        kafkaConfiguration
+            .append(" --override listeners=" + kafkaListeners + "PLAINTEXT://0.0.0.0:" + KAFKA_PORT)
+            .append(" --override advertised.listeners=" + advertisedListeners)
+            .append(" --override zookeeper.connect=localhost:" + ZOOKEEPER_PORT)
+            .append(" --override listener.security.protocol.map=" + kafkaListenerSecurityProtocol + "PLAINTEXT:PLAINTEXT")
+            .append(" --override inter.broker.listener.name=BROKER1");
```

Comment:
You're mixing string concat with `+` and appending to a `StringBuilder`. Just use `append`. Or `String.format` since you're not in a loop.